gnd/src/main/java/com/google/android/gnd/ui/home/featuresheet/RecordListViewModel.java

```diff
@@ -16,32 +16,46 @@
 
 package com.google.android.gnd.ui.home.featuresheet;
 
+import android.util.Log;
+
 import static java8.util.stream.StreamSupport.stream;
 
 import androidx.lifecycle.LiveData;
 import androidx.lifecycle.MutableLiveData;
 import com.google.android.gnd.repository.DataRepository;
+import com.google.android.gnd.rx.Result;
 import com.google.android.gnd.ui.common.AbstractViewModel;
 import com.google.android.gnd.vo.Feature;
 import com.google.android.gnd.vo.Form;
 import com.google.android.gnd.vo.Project;
 import com.google.android.gnd.vo.Record;
 import java.util.Collections;
 import java.util.List;
+
+import io.reactivex.Single;
+import io.reactivex.subjects.PublishSubject;
 import java8.util.Optional;
 import java8.util.stream.Collectors;
 import javax.inject.Inject;
 
 // TODO: Roll up into parent viewmodel. Simplify VMs overall.
 public class RecordListViewModel extends AbstractViewModel {
+
   private static final String TAG = RecordListViewModel.class.getSimpleName();
   private final DataRepository dataRepository;
   private MutableLiveData<List<Record>> recordSummaries;
+  private PublishSubject<RecordSummaryRequest> recordSummaryRequests;
```

Comment:
I feel like the right thing to do here would be for the VM to have access to the active project, feature, and form, either from a shared VM, or from separate streams in this VM. Wdyt? 

If you think it makes sense could you file a TODO against an Issue to fix this in the future?

Reply:
That makes sense to me, added!

Reply:
Though note that the change in commit 0b6ff51 is an alternative. I'm still open to making these items split streams if you think it simplifies things, but RX does provide operators we can use to express the necessary transforms concisely over a single "requests" stream.